File: HRO/CMITDW Taal der Wiskunde/Lua/o53.lua
-- opdracht 5.3; Monte Carlo methode
local p = 15 * 18 -- 15 plaatjes, 18 teams
local t = {}

local function wipe(t)
	for k in pairs(t) do
		t[k] = nil
	end
end

local function MonteCarlo()
	wipe(t)
	local i = 0
	
	while #t ~= p do
		i = i + 1 -- pakjes
		
		for j = 1, 5 do -- 5 plaatjes per pakje
			t[math.random(p)] = true
		end
	end
	
	return i
end

local sum = 0
local n = 100 -- amount of tests

for i = 1, n do
	local result = MonteCarlo()
	sum = sum + result
	print(i, result)
end

print(sum/n.." * 10 euro gemiddeld uitgegeven")

File: HRO/CMITDW Taal der Wiskunde/o53.java
// opdracht 5.3; Monte Carlo methode
import java.util.Random;

public class o53
{
	public static void main(String[] args)
	{
		int sum = 0;
		int n = 100; // amount of tests
		
		for (int i=1; i<=n; i++)
		{
			int result = MonteCarlo();
			sum += result;
			System.out.println(i+"\t"+result);
		}
		
		System.out.println((double) sum/n+" * 10 euro gemiddeld uitgegeven");
	}
	
	public static int MonteCarlo()
	{
		Random rnd = new Random();
		int i = 0;
		boolean[] p = new boolean[15 * 18]; // 15 plaatjes, 18 teams
		
		while (!IsFilled(p))
		{
			i++; // pakjes
			
			for (int j=1; j<=5; j++) // 5 plaatjes per pakje
				p[rnd.nextInt(270)] = true;
		}
		
		return i;
	}
	
	public static boolean IsFilled(boolean b[])
	{
		for (boolean v : b)
		{
			if (!v)
				return false;
		}
		return true;
	}
}

